spec/mongo/collection_spec.rb

```diff
@@ -2000,6 +2000,51 @@ def generate
   end
 
   describe '#count_documents' do
+
+        context 'no argument provided' do
+
+          # no arguments
+          let(:view) do
+            described_class.new(authorized_collection, {}, {})
+          end 
+
+          before do
+            authorized_collection.delete_many
+          end
+
+          context 'empty collection' do
+            it 'returns true' do
+              expect(view.count_documents()).to eq(view.count_documents({}))
+            end
+          end
+
+          context 'nonempty collection' do
+            it 'returns false' do
+              documents = []
+              1.upto(10) do |index|
+                documents << { key: 'a', _id: "in#{index}" }
+              end
+
+              authorized_collection.insert_many(documents)
+              expect(authorized_collection.count_documents()).to eq(10)
+            end
+          end
+
+          context 'nonempty collection' do
+            it 'returns false' do
+              documents = []
+              documents << { key: 'a', _id: "1" }
+              documents << { key: 'b', _id: "2" }
+              documents << { key: 'c', _id: "3" }
+
+              authorized_collection.insert_many(documents)
+              puts authorized_collection.count_documents()
+              expect(authorized_collection.count_documents()).to eq(3)
+            end
+          end
```

Comment:
I don't think you need both this test and the previous test, unless there's some fundamental difference between the case where the collection has 10 elements in it and the one where it only has 3.